cli/index.ts

```diff
@@ -328,6 +333,10 @@ const builtYargs = createYargsCli({
         jsonOutputOption
       ],
       processFn: async argv => {
+        if (!argv.json && !argv.watch) {
```

Comment:
why protect this behind `--json`? that seems quite confusing

Reply:
If you are using --json, you probably are doing something like `dataform compile --json > dump.json`. It would be really annoying if that asked you a question and messed up the stdout!

Reply:
See other comment, resolved.